File: 4th semester/AI (Artificial Intelligence)/Lab6_DecisionTree/Console.py
from Lab6.Controller import Controller
from Lab6.MyException import MyException
import sys


class Console:
    def __init__(self, controller: Controller):
        self.__controller = controller

    def start(self):
        menu = self.__getMenu()
        commands = self.__getCommands()
        while True:
            try:
                print(menu)
                command = input("\t >>> ").strip(" ")
                commands[command]()
            except MyException as exception:
                print(str(exception))
            except KeyError:
                print("Invalid command..")

    def __exitCommand(self):
        print("Closing application..")
        print("Application has been closed.")
        sys.exit()

    def __decisionTreeCommand(self):
        testSizeStr = input("Test size (between 0 and 1):\n\t(0.1 by default)\n\t\t>> ")
        if not testSizeStr:
            testSizeStr = "0.1"
        withShuffleStr = input("With shuffle? (true / false):\n\t(false by default)\n\t\t>> ")
        if not withShuffleStr:
            withShuffleStr = "false"

        self.__controller.run(testSizeStr, withShuffleStr)

    @classmethod
    def __getMenu(self):
        return "1. Decision Tree Algorithm\n" \
               "0. EXIT"

    def __getCommands(self):
        return {"0": self.__exitCommand,
                "1": self.__decisionTreeCommand}

File: 4th semester/SDI (Systems for Design & Implementation)/Exam/exam_9-6-2020/scanner/src/main/java/ro/ubb/scanner/console/Console.java
package ro.ubb.scanner.console;

import org.springframework.beans.factory.annotation.Autowired;
import ro.ubb.scanner.model.ScannerClient;
import ro.ubb.scanner.service.ScannerClientService;

import java.util.HashMap;
import java.util.InputMismatchException;
import java.util.Map;
import java.util.Scanner;

public class Console {
    @Autowired
    private ScannerClientService service;

    public void start() {
        String menu = this.getMenu();
        Map<String, Runnable> commands = this.getCommands();
        String command = "";

        while (!command.equals("0")) {
            System.out.println(menu);

            System.out.println("\t >>> ");
            command = readCommand();

            try {
                commands.get(command).run();
            }
            catch (NullPointerException exception) {
                System.out.println("Invalid command...");
            }
            catch (IllegalArgumentException exception) {
                System.out.println(exception.getMessage());
            }
        }
    }

    private String readCommand() {
        Scanner scanner = new Scanner(System.in);
        return scanner.nextLine();
    }

    private void addScannerCommand() {
        Scanner scanner = new Scanner(System.in);

        String name;
        Integer countyId;

        try {
            System.out.println("Scanner name:\n\t >> ");
            name = scanner.next();

            System.out.println("Scanner county-id (at least 1024) :\n\t >> ");
            countyId = scanner.nextInt();
        }
        catch (InputMismatchException exception) {
            throw new IllegalArgumentException("Invalid integer for county-id");
        }

        if (countyId < 1024) {
            throw new IllegalArgumentException("Id must be greater than 1024");
        }

        ScannerClient scannerClient = ScannerClient.builder()
                .name(name)
                .countyId(countyId)
                .build();

        service.addScanner(scannerClient);
    }

    private void exitCommand() {
        System.out.println("Closing Application...");
        System.out.println("Application has been closed.");
        System.exit(0);
    }

    private String getMenu() {
        StringBuilder menu = new StringBuilder("MENU\n");

        menu.append("1. Add Scanner\n");
        menu.append("0. EXIT\n");

        return menu.toString();
    }

    private Map<String, Runnable> getCommands() {
        Map<String, Runnable> commands = new HashMap<>();
        commands.put("0", this::exitCommand);
        commands.put("1", this::addScannerCommand);
        return commands;
    }
}
